backend/spec/controllers/spree/admin/products_controller_spec.rb

```diff
@@ -50,6 +50,23 @@
     end
   end
 
+  # regression test for https://github.com/spree/spree/issues/2791
+  context "adding taxons to a product" do
+    let!(:product) { create(:product) }
+    let!(:first_taxon) { create(:taxon) }
+    let!(:second_taxon) { create(:taxon) }
```

Comment:
This taxon is needlessly created for the first spec. `let!` can also be problematic in creating records when you don't need them and can slow test suites down. Using `let` will solve this.

Reply:
Thanks for the feedback @BenMorganIO, changes made. Let me know if there is anything else you can think of!